src/Controller/Traits/PasswordManagementTrait.php

```diff
@@ -133,4 +133,54 @@ public function requestResetPassword()
             $this->Flash->error(__d('CakeDC/Users', 'Token could not be reset'));
         }
     }
+
+    /**
+     * resetGoogleAuthenticator
+     *
+     * Resets Google Authenticator token by setting secret_verified
+     * to false.
+     * Currently allowed to be done either by superuser or user himself.
+     *
+     * @param mixed $id of the user record.
+     * @return mixed.
+     */
+    public function resetGoogleAuthenticator($id = null)
+    {
+        $allowed = false;
+        $currentUser = !empty($this->Auth->user()) ? $this->Auth->user() : false;
+
+        if (!$currentUser) {
+            $message = __d('CakeDC/User', 'Please login to the system');
+            $this->Flash->error($message, 'auth');
+
+            return $this->redirect($this->Auth->config('loginAction'));
+        }
+
+        if (true == $currentUser['is_superuser'] || $id == $currentUser['id']) {
```

Comment:
we should manage access to this action via permissions, no need to add extra logic to do it here